src/lang/parser.mly

```diff
@@ -397,56 +380,20 @@ if_elsif:
   | ELSE exprs                      { mk_fun ~pos:($startpos($1),$endpos($2)) [] $2 }
   |                                 { mk_fun ~pos:$loc [] (mk ~pos:$loc unit) }
 
-
-app_opt:
+encoder_opt:
   | %prec no_app { [] }
-  | LPAR app_list RPAR { $2 }
-
-top_level_ogg_audio_item:
-  | VORBIS app_opt     { Lang_vorbis.make $2 }
-  | VORBIS_CBR app_opt { Lang_vorbis.make_cbr $2 }
-  | VORBIS_ABR app_opt { Lang_vorbis.make_abr $2 }
-  | SPEEX app_opt      { Lang_speex.make $2 }
-  | OPUS app_opt       { Lang_opus.make $2 }
-
-ogg_audio_item:
-  | FLAC app_opt             { Lang_flac.make_ogg $2 }
-  | top_level_ogg_audio_item { $1 }
-
-ogg_video_item:
-  | THEORA app_opt     { Lang_theora.make $2 }
-
-ffmpeg_param:
-  | STRING GETS expr     { $1,$3 }
-  | VAR GETS expr        { $1,$3 }
-ffmpeg_params:
-  |                                  { [] }
-  | ffmpeg_param                     { [$1] }
-  | ffmpeg_param COMMA ffmpeg_params { $1::$3 }
-
-ffmpeg_list_elem:
-  | AUDIO_NONE                        { `Audio_none }
-  | AUDIO_COPY                        { `Audio_copy }
-  | AUDIO_RAW LPAR ffmpeg_params RPAR { `Audio_raw $3 }
-  /* This is for inline encoders. */
-  | AUDIO_RAW                         { `Audio_raw [] }
-  | AUDIO LPAR ffmpeg_params RPAR     { `Audio  $3 }
-  | VIDEO_NONE                        { `Video_none }
-  | VIDEO_COPY                        { `Video_copy }
-  /* This is for inline encoders. */
-  | VIDEO_RAW                         { `Video_raw [] }
-  | VIDEO_RAW LPAR ffmpeg_params RPAR { `Video_raw  $3 }
-  | VIDEO LPAR ffmpeg_params RPAR     { `Video  $3 }
-  | ffmpeg_param                      { `Option $1 }
- 
-ffmpeg_list:
-  |                                    { [] }
-  | ffmpeg_list_elem                   { [$1] }
-  | ffmpeg_list_elem COMMA ffmpeg_list { $1::$3 }
+  | LPAR encoder_params RPAR { $2 }
 
-ffmpeg_opt:
-  | %prec no_app { [] }
-  | LPAR ffmpeg_list RPAR { $2 }
+encoder_param:
+  | VAR GETS expr       { $1, `Term $3 }
+  | STRING GETS expr    { $1, `Term $3 }
+  | VAR                 { "", `Term (mk ~pos:$loc (Ground (String $1))) }
+  | ENCODER encoder_opt { "", `Encoder ($1, $2) }
```

Comment:
Maybe also add `STRING` in case we want to do `%encoder("some non var parameter")`.